tests/test_uhura.py

```diff
@@ -56,34 +39,44 @@ def test_uhura_file_and_s3(test_df, file_resource, s3_resource, tmpdir):
         with pytest.raises(AssertionError):
             file_resource.write(df.drop("a", axis=1))
 
-    # Write for s3 test
-    file_resource.path = Path(tmpdir / "uhura" / "input" / "s3" / "bucket" / "some_file.extension")
+
+@pytest.fixture
+def s3_fixtures(file_name, tmpdir, test_df):
+    # Fixtures setup for s3 test
+    file_resource = LocalFileResource(path=tmpdir / "actual" / file_name)
+    file_resource.path = Path(tmpdir / "uhura" / "input" / "s3" / "bucket" / file_name)
     file_resource.write(test_df)
-    file_resource.path = Path(tmpdir / "uhura" / "output" / "s3" / "bucket" / "some_file.extension")
+    file_resource.path = Path(tmpdir / "uhura" / "output" / "s3" / "bucket" / file_name)
     file_resource.write(test_df)
 
+
+def test_uhura_s3(test_df, tmpdir, file_name, s3_fixtures):
+    # Actual test
+    s3_resource = S3Resource(bucket="bucket", path=file_name)
     with task_test_mode(input_path=tmpdir / "uhura" / "input", known_good_path=tmpdir / "uhura" / "output"):
         df = s3_resource.read()
         s3_resource.write(df)
-        with pytest.raises(AssertionError):
-            s3_resource.write(df.drop("a", axis=1))
-
 
-def test_postgres_uhura(tmpdir, test_df):
-    postgres_resource = PostgresResource(db_user="asdf", db_host="asdf", db_name="asdf", table_name="tabular_table")
-    ParquetResource(path=tmpdir / "uhura" / "input" / "postgres" / "tabular_table.parquet").write(test_df)
-    ParquetResource(path=tmpdir / "uhura" / "output" / "postgres" / "tabular_table.parquet").write(test_df)
-    with task_test_mode(input_path=tmpdir / "uhura" / "input", known_good_path=tmpdir / "uhura" / "output"):
-        postgres_resource.read()
-        postgres_resource.write(test_df)
+        # Check that, in test mode, the dfs are being compared and if not the same -> fail.
         with pytest.raises(AssertionError):
-            postgres_resource.write(test_df.drop("a", axis=1))
+            s3_resource.write(df.drop("a", axis=1))
 
 
-def test_kafka_uhura(tmpdir, test_df):
-    kafka_resource = KafkaResource(topic="tropico", server="asdf")
-    ParquetResource(path=tmpdir / "uhura" / "output" / "kafka" / "tropico").write(test_df)
-    with task_test_mode(input_path=tmpdir / "uhura" / "input", known_good_path=tmpdir / "uhura" / "output"):
-        kafka_resource.write(test_df)
-        with pytest.raises(AssertionError):
-            kafka_resource.write(test_df.drop("a", axis=1))
+# def test_postgres_uhura(tmpdir, test_df):
```

Comment:
remove these?

Reply:
I should probably put these tests back in.

Reply:
I added these tests back in - they also revealed that I should add file extensions to the auto-generated fixture paths for postgres and kafka(even though not strictly necessary they will make life easier.)